utils/device_pool.py

```diff
@@ -1,107 +1,211 @@
 #!/usr/bin/env python
-from lockfile import LockFile, NotLocked
 import os
 import logging
 import subprocess
+import re
+import socket
+import traceback
 
+from mozdevice.devicemanager import DMError
+from lockfile import LockFile, NotLocked
 
 logging.basicConfig(level=logging.INFO)
 logger = logging.getLogger(__name__)
 
 
-class DevicePool(object):
-    lock_folder = "/tmp/LOCKS"
-    my_lock = None
-    serial = None
-    current_lock_index = -1
-
-    def __init__(self, deviceSerial=None):
-        self.devices = self._device_list()
-        if deviceSerial:
-            self.serial = deviceSerial
-            self._check_serial_in_devices()
-        if not os.path.exists(self.lock_folder):
-            os.makedirs(self.lock_folder)
-        self.lock_list = self._get_lock_list()
-
-    def _check_serial_in_devices(self):
-        if self.serial:
-            chk_list = filter(lambda x: x == self.serial, self.devices)
-            if not chk_list:
-                logger.warning("Android serial[" + self.serial + "] can't be found")
-                return False
-            else:
-                return True
+def exec_process(cmd_list):
```

Comment:
the main purpose is to close the stdin, stdout and stderr obj, I found some ppl are discussion the OSError 24 is caused by not closing these objs